modules/rostests/apitests/ntdll/NtQueryInformationProcess.c

```diff
@@ -323,6 +325,103 @@ Test_ProcessWx86Information(void)
     trace("VdmPower = %lu\n", VdmPower);
 }
 
+static
+void
+Test_ProcQueryAlignmentProbe(void)
+{
+    NTSTATUS Status;
+    ULONG InfoClass;
+
+    /* Iterate over the process info classes and begin the tests */
+    for (InfoClass = 0; InfoClass < _countof(PsProcessInfoClass); InfoClass++)
+    {
+        /* The buffer is misaligned */
+        Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                           InfoClass,
+                                           (PVOID)(ULONG_PTR)(PsProcessInfoClass[InfoClass].AlignmentQUERY / 2),
+                                           PsProcessInfoClass[InfoClass].RequiredSizeQUERY,
+                                           NULL);
+
+        /* Ensure that we don't get any of these status values */
+        if (Status != STATUS_INVALID_INFO_CLASS &&
+            Status != STATUS_INFO_LENGTH_MISMATCH &&
+            Status != STATUS_UNSUCCESSFUL)
```

Comment:
These values are expected only for the classes you call out in the `switch` below, right?

Doing the check this way around unfortunately means you would miss something like `ProcessBasicInformation` being totally broken and always failing with `STATUS_UNSUCCESSFUL` (it would ignore the check here and isn't present in the switch, so you'd check nothing at all).

So checking the other way around may be nicer, e.g.:
```
if (!(PsProcessInfoClass[InfoClass].Flags & ICIF_QUERY))
{
    ok(Status == STATUS_INVALID_INFO_CLASS);
}
else
{
    ok(Status == STATUS_DATATYPE_MISALIGNMENT);
}
```

Or if you really need the switch, perhaps move it to the front, e.g.
```
NTSTATUS SpecialStatus = STATUS_SUCCESS;
switch (...)
{
    case Xxx:
        SpecialStatus = STATUS_UNSUCCESSFUL;
}

if (SpecialStatus)
    ok(Status == SpecialStatus);
else
    ok(Status == STATUS_DATATYPE_MISALIGNMENT);
```

Reply:
If XP behaves really weirdly for some classes (i.e. something we would consider a bug if ROS did it), then I'd rather have the test fail in XP than miss a bug in ROS.